### PR TITLE
Update super-linter.yml to use super-linter/super-linter

### PR DESCRIPTION
**Pull request by echoix**
Super-linter is now in its own organization since April 26th, 2023. This updates the workflow to use newer releases.

### CODE REVIEW

**Comment by echoix** [comment]
I’m not particularly a big fan of super-linter, the way it is can be configured, but I’m obviously biased since I’ve been collaborator and maintainer of MegaLinter for two years now, a project that had its roots in super-linter. But this workflow still obviously deserves to be maintained and updated, even if some further changes might do otherwise. It’s our « now » tool, so we need to keep it working correctly :)

**Comment by echoix** [comment]
I’ll come back with the fixes tomorrow, I remember fixing them in one of my exploratory branches, I just have to find them.

**Comment by echoix** [comment]
Ready to review

**Review by wenzeslaus — APPROVED** [review]
+1 for the update, but I'm a fan of MegaLinter!

**Comment by echoix** [comment]
> +1 for the update, but I'm a fan of MegaLinter!

I thought the direction we were going was to use precommit ci (#3255)

I was holding off on optimizing the h*ll out of our linters, since the repo had another vision/direction that I respect and will play with. It is also the reason I didn't push further on #3284, and didn't complete it, since it wasn't really wanted, and the general attitude here is to have everything right the first time, thus waiting and leaving out smaller improvements, rather than continuously improve by small steps. Both opinions are justified, it only needs to get used to, and I respect both.

**Comment by wenzeslaus** [comment]
> > +1 for the update, but I'm a fan of MegaLinter!
> 
> I thought the direction we were going was to use precommit ci (#3255)

I think there is a value in having multiple tools. One does not respect the settings (like Flake8 in pre-commit), another one is not used locally (usually true for MegaLinter, super-linter), ... Also when there is a problem with one, the other is still in place (we can disable one and continue, instead of completely migrating from one to another).

**Pull request merged**
